crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -1097,28 +1100,47 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
             Flow::Local { .. } => {
                 trace!("Adding new local timeline item");
 
-                let item = self.meta.new_timeline_item(item);
+                let item = Self::new_timeline_item(self.meta, item, None);
```

Comment:
Since we statically know there's no replacement, can we keep on using `self.meta.new_timeline_item`, to avoid doing unnecessary work in `Self::new_timeline_item`, and avoid one layer of indirection, please?

Reply:
I was more inclined of having a consistency here and having a single way to create a `TimelineItem` from an `EventTimelineItem`, but your argument is acceptable. Maybe adding `#[inline(always)]` would help the compiler to inline and remove the branch with `None` here since it's statically known, but well.

I've updated the code to `self.meta.new_timeline_item`.